diff --git a/api/opentrons/drivers/smoothie_drivers/v2_0_0/virtual_smoothie.py b/api/opentrons/drivers/smoothie_drivers/v2_0_0/virtual_smoothie.py
index a6cda3d..1557054 100644
--- a/api/opentrons/drivers/smoothie_drivers/v2_0_0/virtual_smoothie.py
+++ b/api/opentrons/drivers/smoothie_drivers/v2_0_0/virtual_smoothie.py
@@ -1,4 +1,5 @@
 import io
+import os
 import re
 
 from opentrons.drivers.smoothie_drivers import VirtualSmoothie
@@ -343,6 +344,11 @@ class VirtualSmoothie_2_0_0(VirtualSmoothie):
                 #     'Command {} is not supported'.format(command))
 
     def write(self, data):
+        gfile = os.environ.get('GCODE_FILE')
+        if gfile:
+            with open(gfile, 'a') as gf:
+                gf.write(data.decode())
+
         if not self.isOpen():
             raise RuntimeError('Virtual Smoothie not currently connected')
         if not isinstance(data, str):
